src/main/java/headcube/Storage.java

```diff
@@ -0,0 +1,116 @@
+package headcube;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+/**
+ * Handles storage operations for the HeadCube application, including saving and loading tasks.
+ */
+public class Storage {
+    private Ui ui;
+
+    /**
+     * Constructor a Storage object with a reference to the UI for displaying messages.
+     *
+     * @param ui The UI to display messages.
+     */
+    public Storage(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Saves the current list of tasks to a file.
+     *
+     * @param taskList The list of tasks to be saved.
+     * @return The save message
+     */
+    public String save(TaskList taskList) {
+        try {
+            String directoryPath = "./data";
+            String filePath = directoryPath + "/HeadCube.txt";
```

Comment:
Consider storing these as fields, would be helpful to allow access to the filePath in the `load` method below